aten/src/ATen/TensorImpl.cpp

```diff
@@ -62,9 +62,8 @@ void Tensor::backward(
 TensorImpl::TensorImpl(TensorTypeId type_id, ScalarType scalar_type)
     : type_id_(type_id), scalar_type_(scalar_type) {
   auto type = &globalContext().getType(tensorTypeIdToBackend(type_id), scalar_type);
-  Storage* storage = type->storage(true).release();
-  StorageImpl* storage_impl = storage->pImpl();
-  tensor = new THTensor(storage_impl);
+  Storage storage = type->storage(true);
+  tensor = new THTensor(storage.release());
```

Comment:
this is going away anyway, but shouldn't you release after you pass in to the tensor?

Reply:
I guess it technically doesn't matter, but it looks scary.

Reply:
Actually, there is something really scary here which we really should fix. In this context, `release` means, "relinquish ownership of the pointer managed by this smart pointer, and return the raw pointer" (which is immediately reclaimed by the `THTensor` here). However, on StorageImpl, we have `_raw_release`, which means "decrement this refcount". Probably should just rename `_raw_retain` and `_raw_release` to `_raw_incref` and `_raw_decref`.